### ISSUE TITLE
[unbound-method] `this: void` syntax should be highlighted in document and suggested in message

### ISSUE THREAD

**Issue by JounQin** [typescript-eslint/typescript-eslint#3201]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

https://github.com/sveltejs/eslint-plugin-svelte3/issues/103#issuecomment-802415147

```json
{
  "rules": {
    "@typescript-eslint/unbound-method": 2
  }
}
```

```ts
// on:animationend={myStore.action}

const myStore = {
  action(event: AnimationEvent | string) {
    // handle event here
  },
}
```

**Expected Result**

Better message and better document for suggesting `this: void`

**Actual Result**

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.10.0` |
| `@typescript-eslint/parser`        | `4.10.0` |
| `TypeScript`                       | `4.2.3` |
| `ESLint`                           | `7.10.0` |
| `node`                             | `14.16.0` |
